datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/load/BatchGetEntitiesResolver.java

```diff
@@ -49,9 +61,32 @@ public CompletableFuture<List<Entity>> get(DataFetchingEnvironment environment)
 
     return CompletableFuture.allOf(entitiesFutures.toArray(new CompletableFuture[0]))
         .thenApply(
-            v ->
-                entitiesFutures.stream()
-                    .flatMap(future -> future.join().stream())
-                    .collect(Collectors.toList()));
+            v -> {
+              Entity[] finalEntityList = new Entity[entities.size()];
+              // Returned objects can be either of type Entity or wrapped as
+              // DataFetcherResult<Entity>
+              // Therefore we need to be working with raw Objects in this area of the code
```

Comment:
hmm why is this the case that it could be returned as one thing or another? locally my IDE believes that it should reliably return a list of `Entity` when we flatmap the entitiesFutures.

not a real blocker for me since your code works well for handling the alternate case but if we didn't have to have that that would be better

Reply:
Yeah this one stems from an incorrect assertion that the underlying BatchUtils method is making. Changing that signature would have created a cascade of changes that I wanted to avoid in this PR.